### PR TITLE
feat: activate and deactivate are bulk publish controls; add unpublished option for previewing

### PR DESCRIPTION
**Pull request by dkoo**
Instead of using an "active" group, campaigns are displayed on the front-end if they are published and if they belong to a segment matching the current reader.

### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Requires `newspack-popups` be on `feat/current-campaign-group-pub-status`, see: https://github.com/Automattic/newspack-popups/pull/375

Like in #775, introduces the concept of bulk-activating and deactivating campaigns by group. Unlike #775, this branch uses the WP post status of campaigns to determine whether or not individual campaigns should be shown on the front-end, and turns "Activate" and "Deactivate" buttons into bulk publish/unpublish actions that apply to the currently viewed campaigns. "Activating" campaigns when viewing a particular group of campaigns will publish any unpublished campaigns in that group and also automatically unpublish all campaigns that don't belong to the viewed group. This results in an editor experience very similar to that of #775, but without requiring the additional "active group" status for campaigns to appear on the front-end. This also avoids an expensive tax query to fetch campaigns on the front-end, instead relying on post IDs only.

Also adds a new option in the preview UIs to let users choose whether to show unpublished campaigns. If checked, the preview modal will show draft, pending, scheduled and published campaigns. If unchecked, only published campaigns will be shown (matching the front-end display).

Screen Shot 2021-01-08 at 3 12 05 PM

~Note that for test purposes I've removed labels for Test and "Never" frequency campaigns and ignored those criteria when displaying campaigns in anticipation of #776, but I did not fully deprecate all functionality for those statuses here.~ I've restored these for now so that we can more cleanly deprecate them in the other PR.

### How to test the changes in this Pull Request:

1. View the Campaigns tab in the wizard. Confirm that the filter always defaults to the "All campaigns" filter view.
2. Confirm that published campaigns are shown with a green label and unpublished campaigns (draft, pending or scheduled) are shown with a grey label.
3. Expand the popover menu for a published campaign. Confirm that there's a new "Unpublish" option that when clicked, reverts the selected campaign to draft status.
4. Observe the "Publish All", "Unpublish All" and "Preview" buttons that are now visible in the All Campaigns filter view.
5. Click "Publish All" and confirm that this sets ALL campaigns to "publish" status. Then click "Unpublish All" and confirm that it sets ALL campaigns to "draft" status.
6. Filter the campaigns list by a group. Click "Publish All" and confirm that this sets only the campaigns in this group to "publish" status. (Make sure to view another group to confirm that those campaigns are still in draft status).
7. View another group and click "Publish All". Confirm that the campaigns in this group are now set to "publish" status, and campaigns in the first group that don't exist in this second group are set to "draft" status.
8. Publish some campaigns from various groups. Then filter by one group and click "Unpublish All." Confirm that only the campaigns in the filtered group are unpublished.
9. In the preview popover, observe a new checkbox option to "Show unpublished campaigns". Preview various groups and segments and confirm that preview still works as expected. Also confirm that draft, pending, scheduled and published campaigns are all shown in the preview modal if the "Show unpublished campaigns" option is selected. Repeat these tests with both the Campaigns and Preview tabs.
10. View the site front-end in a new incognito window and confirm that only published campaigns matching the reader's segment are shown.
11. The one caveat to (10) is that campaigns that are set to "Uncategorized" are now able to show with any category—this seems more intuitive to me, especially since uncategorized campaigns are automatically assigned the "Uncategorized" category when published.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by thomasguillot** [comment]
@dkoo in https://github.com/Automattic/newspack-plugin/pull/799/commits/a982bb0576f0f1531537b8b119b2f9476333a20b and https://github.com/Automattic/newspack-plugin/pull/799/commits/ab970b4b96053342a462f6949541cc2bb221fcce I pushed a few visual changes to your PR. I thought it was better to do it directly here rather than creating a new PR later on.

**Review by jeffersonrabb — CHANGES_REQUESTED** [review]
This is working well - left a few comments.

File: assets/wizards/popups/index.js [hunk 1]
```diff
@@ -200,13 +218,16 @@ class PopupsWizard extends Component {
 		return `${ previewURL }?${ stringify( { ...options, newspack_popups_preview_id: id } ) }`;
 	};
 
-	manageCampaignGroup = ( id, method = 'POST' ) => {
+	manageCampaignGroup = ( campaigns, method = 'POST' ) => {
 		const { setError, wizardApiFetch } = this.props;
+		const campaignIds = campaigns.map( campaign => campaign.id );
 		return wizardApiFetch( {
-			path: `/newspack/v1/wizard/newspack-popups-wizard/campaign-group/${ id }`,
+			path: addQueryArgs( '/newspack/v1/wizard/newspack-popups-wizard/campaign-group/', {
```
[the remark this answers is not in the document]
Comment (dkoo) [reply assets/wizards/popups/index.js]: Good idea, updated in 85f2335.

File: includes/wizards/class-popups-wizard.php [hunk 2]
```diff
@@ -288,7 +305,7 @@ public function register_api_endpoints() {
 
 		register_rest_route(
 			NEWSPACK_API_NAMESPACE,
-			'/wizard/' . $this->slug . '/campaign-group/(?P<id>\d+)',
+			'/wizard/' . $this->slug . '/campaign-group',
```
[the remark this answers is not in the document]
Comment (dkoo) [reply includes/wizards/class-popups-wizard.php]: Updated in 85f2335.

File: includes/wizards/class-popups-wizard.php [hunk 3]
```diff
@@ -177,6 +177,23 @@ public function register_api_endpoints() {
 				],
 			]
 		);
+		register_rest_route(
```
[the remark this answers is not in the document]
Comment (dkoo) [reply includes/wizards/class-popups-wizard.php:180]: Updated in 85f2335.

**Review by jeffersonrabb — APPROVED** [review]
Looks good! You'll have some merge conflicts, and there are a couple small bits you'll be able to remove (e.g. the localstorage part of previewing). Also, I'd suggest renaming `activate_campaign_group` and `deactivate_campaign_group` everywhere (configuration manager and `newspack-popups`) just to avoid confusion, but that won't require another review.

**Comment by dkoo** [comment]
Renamed the `activate_campaign_group` and `deactivate_campaign_group` functions and fixe conflicts in 56562bb. Thanks!

**Pull request merged**